### ISSUE TITLE
Low accuracy on private dataset

### ISSUE THREAD

**Issue by WIll-Xu35** [auroua/InsightFace_TF#57]
I tested model d on my own data set. Compare with mxnet implementation of insightface, which have accuracy of 99.8%, model d can only reach 96%. I'm wondering if I did something wrong.

I modified the eval_ckpt_file.py and make it generate embedding for one image at a time. I simply feed the model with a aligned image, get the embedding, compare the distance and make a decision.

I suspect there might be something wrong during the mtcnn detection part. However I tried margin value 44, 10, 0 and none of those configuration bring a huge improvement.

Any help or discussion is appreciated. Thanks!

**Comment by andyguo007** [comment]
Did you normalize you data like below before feeding to the model?

img -= 127.5
img *= 0.0078125

**Comment by WIll-Xu35** [comment]
@andyguo007 Thanks, I do forgot to normalize the input. Now I have acc of 99.3%, not bad but still has some performance drop compared with the pretrained model of the original implementation.

**Comment by andyguo007** [comment]
@WIll-Xu35 the peformance drop might be due to the use of a smaller batch size (assuming you compare model d with LResNet50E in the original implementation already).

Did you try different face alignment methods to compare the performance difference?, i.e., 

M = cv2.estimateRigidTransform( dst.reshape(1,5,2), src.reshape(1,5,2), False)
warped = cv2.warpAffine(img,M,(shape[1],shape[0]), borderValue = 0.0)

and 

tform = trans.SimilarityTransform()                                                                                                                                                  
tform.estimate(dst, src)
M = tform.params[0:2,:]
warped = cv2.warpAffine(img,M,(shape[1],shape[0]), borderValue = 0.0)

**Comment by WIll-Xu35** [comment]
@andyguo007 I do compared model d with LResNet50E in the original implementation. So the batch size used in the original implementation is larger and may have a better performance?

I always use the second method you mentioned to do face alignment because I found it in the original implementation. It gives a really good result so I didn't even think about changing it. I'll try to test the other alignment method when I have time. If you have already done so, could you please share the results?

Many thanks.

**Comment by andyguo007** [comment]
@WIll-Xu35 larger batch size possibly generates better results.

I didn't try estimateRigidTransform either. SimilarityTransform was used in training. To be consistent, it makes sense to use SimilarityTransform in testing as well. It is said that different alignment methods have a big impact on the performance. Wondering how "big" the impact is when we use estimateRigidTransform in training but SimilarityTransform in testing.

**Comment by WIll-Xu35** [comment]
@andyguo007 if I remember correctly, I used to use insightface without alignment and tested it on my private dataset, the accuracy is around 96%. After I applied face alignment correctly, the accuracy reached 99.8%.
